### PR TITLE
Fixes some assembly runtimes

### PR DESCRIPTION
**Pull request by Ahbahl**
Fixes null.density runtime from infrared.
Fixes null.key runtime from heartmonitor-signaler assemblies.
Adds ismonitor() to helpers.dm
Adds better logging to signaler logs.

### CODE REVIEW

File: code/modules/assembly/helpers.dm [hunk 1]
```diff
@@ -28,6 +28,10 @@
 		return 1
 	return 0
 
+/proc/ismonitor(O)
```
Comment (TheDracheX) [inline code/modules/assembly/helpers.dm]: I feel like this is unnecessary and ambigious. Why is it needed?

File: code/modules/assembly/signaler.dm [hunk 2]
```diff
@@ -109,7 +109,12 @@
 
 		var/time = time2text(world.realtime,"hh:mm:ss")
 		var/turf/T = get_turf(src)
-		lastsignalers.Add("[time] <B>:</B> [usr.key] used [src] @ location ([T.x],[T.y],[T.z]) <B>:</B> [format_frequency(frequency)]/[code]")
+		if(usr)
+			lastsignalers.Add("[time] <B>:</B> [usr.real_name]/([usr.key]) triggered [holder ? holder.name : src] @ location ([T.x],[T.y],[T.z]) <B>:</B> [format_frequency(frequency)]/[code]")
+		if(!usr && holder)
+			if(istype(holder.loc, /mob/living))
+				var/mob/living/L = holder.loc
+				lastsignalers.Add("[time] <B>:</B> [L.real_name]/([L.key]) triggered [holder.name] @ location ([T.x],[T.y],[T.z]) <B>:</B> [format_frequency(frequency)]/[code]")
```
Comment (TheDracheX) [inline code/modules/assembly/signaler.dm]: Use key_name_admin for displaying a users key/character name

File: code/modules/assembly/heartmonitor.dm [hunk 3]
```diff
@@ -15,4 +15,7 @@
 		if(istype(holder.loc, /mob/living))
 			var/mob/living/L = holder.loc
 			if(L.stat || L.stunned || L.weakened)
-				pulse()
```
Comment (TheDracheX) [inline code/modules/assembly/heartmonitor.dm]: What's this do?

File: code/modules/assembly/helpers.dm (hunk 1 above)
Comment (Ahbahl) [inline code/modules/assembly/helpers.dm]: Its not necessary but all the other assemblies had an is() proc so I thought why not?

File: code/modules/assembly/heartmonitor.dm (hunk 3 above)
Comment (Ahbahl) [inline code/modules/assembly/heartmonitor.dm]: Fragments of when I was trying to figure out how the signaler magically gets the usr from the other assemblies.

**Pull request merged**